### ISSUE TITLE
Couldnt find http-server

### ISSUE THREAD

**Issue by Nukulargamer** [pierre-josselin/payday-2-ultimate-trainer-6#25]
PAYDAY 2 - Ultimate Trainer 6

--- TEST ---

You must close this window before using the mod!

Checking Node.js version... OK
Checking installation status... OK
Testing application web server...

An error has occurred:

'http-server' is not recognized as an internal or external command,
operable program or batch file.

**Comment by pierre-josselin** [comment]
Hi, thank you for your feedback.

Can you try this:

- Close the game and the consoles
- Open `C:\Program Files (x86)\Steam\steamapps\common\PAYDAY 2` in windows explorer
- Right-click _payday2_win32_release.exe_ and click _Run as administrator_

Please let me know the result, then I could offer you a lasting solution.

**Comment by Nukulargamer** [comment]
Running payday 2 as administrator resulted in the same error

**Comment by pierre-josselin** [comment]
Thank you. This should fix the error:

- Download [test.bat](https://drive.google.com/file/d/1CaoLWXM4dtRejjpBCxIPAJhswiAB9Tc3/view?usp=sharing)
- **Right-click the file and click on "Run as administrator"**
- You will probably get a Microsoft SmartScreen warning, click "more info" then "Run anyway"
- Try the mod again

Note: this script will add **npm** and **http-server** paths to user and system env variables, here is the source (you can also open the file with any text editor):

```bat
@echo off
setlocal

NET SESSION >nul 2>&1
IF NOT %ERRORLEVEL% EQU 0 (
    echo Please run this script as admin.
    pause
    exit /b
)

set "npmPath=%APPDATA%\npm"
set "httpServerPath=%APPDATA%\npm\node_modules\http-server"

for /f "usebackq delims=" %%a in (`powershell -command "[Environment]::GetEnvironmentVariable('Path', 'User')"`) do set "userPath=%%a"
for /f "usebackq delims=" %%a in (`powershell -command "[Environment]::GetEnvironmentVariable('Path', 'Machine')"`) do set "systemPath=%%a"

echo %userPath% > user-path-backup.txt
echo %systemPath% > system-path-backup.txt

set "newUserPath=%userPath%;%npmPath%;%httpServerPath%"
set "newSystemPath=%systemPath%;%npmPath%;%httpServerPath%"

powershell -command "[Environment]::SetEnvironmentVariable('Path', '%newUserPath%', 'User')"
powershell -command "[Environment]::SetEnvironmentVariable('Path', '%newSystemPath%', 'Machine')"

echo Successfully added paths.

pause
```

I recommend you to first backup your user and system env variables.

Please let me know if it worked!

**Comment by Nukulargamer** [comment]
Okay, That worked. Now becuase my computer is named Tegan&Kayla Desktop its causing issues Below is the test.txt


--- TEST ---

You must close this window before using the mod!

Checking Node.js version... OK
Checking installation status... OK
Testing application web server...

An error has occurred:


'Kayla' is not recognized as an internal or external command,
operable program or batch file.
node:internal/modules/cjs/loader:1080
  throw err;
  ^

Error: Cannot find module 'C:\Users\Tegan\node_modules\http-server\bin\http-server'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1077:15)
    at Module._load (node:internal/modules/cjs/loader:922:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:81:12)
    at node:internal/main/run_main_module:23:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v18.17.0

**Comment by pierre-josselin** [comment]
Hi, thanks for your response.

So if the problems are caused by a special character in your username, installing http-server directly into the project folder should fix the problem. Can you try:

Extract all the contents of the attached archive into `payday-2-ultimate-trainer-6`

[http-server-fix.zip](https://github.com/pierre-josselin/payday-2-ultimate-trainer-6/files/12342027/http-server-fix.zip)

Like this:

![image](https://github.com/pierre-josselin/payday-2-ultimate-trainer-6/assets/70814177/c65b046c-5030-4fe7-a6b5-05dd029233bc)

**Comment by Nukulargamer** [comment]
Can conform this fixed the issue.

PAYDAY 2 - Ultimate Trainer 6

--- TEST ---

You must close this window before using the mod!

Checking Node.js version... OK
Checking installation status... OK
Testing application web server... OK
Testing server... OK

All tests passed!
You can now close this window.

**Comment by pierre-josselin** [comment]
Glad to hear it, I fixed the issue for 6.3.0 with [this commit](https://github.com/pierre-josselin/payday-2-ultimate-trainer-6/commit/3791eae472e6531d664672fd2ae1b9e7c43d8ef9).

Feel free to open a new issue if you have any other problems.

**Comment by Nukulargamer** [comment]
Sounds good, thank you for the assist and hopefully this helps others!
